src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.h

```diff
@@ -125,27 +127,53 @@ class Delegate
+     *   @param  powerSettingNum: the input power setting value.
+     *   @param  wattSettingIndex: the input watts setting index.
+     *   Note: powerSettingNum and wattSettingIndex must be mutually exclusive.
+     *   If using power as number, wattSettingIndex will be set to NullOptional.
+     *   If using power in watts, powerSettingNum will be set to NullOptional.
      */
-    virtual Protocols::InteractionModel::Status HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime,
-                                                                                   uint8_t powerSetting) = 0;
+    virtual Protocols::InteractionModel::Status HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTimeSec,
+                                                                                   bool startAfterSetting,
+                                                                                   Optional<uint8_t> powerSettingNum,
+                                                                                   Optional<uint8_t> wattSettingIndex) = 0;
 
     /**
-     *   @brief Handle Command Callback in application: AddMoreTime
-     *   @return Returns the Interaction Model status code which was user determined in the business logic
-     *   @param  finalCookTime: the cook time value after adding input time
+     *   @brief Handle Command Callback in application: AddMoreTime.
+     *   @return Returns the Interaction Model status code which was user determined in the business logic.
+     *   @param  finalCookTimeSec: the cook time value after adding input time.
      */
-    virtual Protocols::InteractionModel::Status HandleModifyCookTimeCallback(uint32_t finalCookTime) = 0;
+    virtual Protocols::InteractionModel::Status HandleModifyCookTimeSecondsCallback(uint32_t finalCookTimeSec) = 0;
+
+    /**
+     *   Get the watt setting from the supported watts list.
+     *   @param index The index of the watt setting to be returned. It is assumed that watt setting are indexable from 0 and with no
+     * gaps.
+     *   @param wattSetting A reference to receive the watt setting on success.
+     *   @return Returns a CHIP_NO_ERROR if there was no error and the label was returned successfully,
+     *   CHIP_ERROR_NOT_FOUND if the index in beyond the list of available labels.
+     */
+    virtual CHIP_ERROR GetWattSettingByIndex(uint8_t index, uint16_t & wattSetting) = 0;
+
+    virtual uint32_t GetMaxCookTimeSec() const = 0;
+
+    virtual uint8_t GetPowerSettingNum() const = 0;
+
+    virtual uint8_t GetMinPowerNum() const = 0;
+
+    virtual uint8_t GetMaxPowerNum() const = 0;
 
-    virtual uint8_t GetMinPower() const = 0;
+    virtual uint8_t GetPowerStepNum() const = 0;
 
-    virtual uint8_t GetMaxPower() const = 0;
+    virtual uint8_t GetCurrentWattIndex() const = 0;
 
-    virtual uint8_t GetPowerStep() const = 0;
+    virtual uint16_t GetCurrentWattRating() const = 0;
```

Comment:
The watt rating is fixed.  Recommend changing this name to `GetWattRating`.

```suggestion
    virtual uint16_t GetWattRating() const = 0;
```